### DISCUSSION TITLE
Stencil Search

### DISCUSSION THREAD

**Discussion by gordon-brown** [clientIO/joint#2607] [Q&A]
### Introduction

Currently when you search a stencil. It will show matching shapes when the palette is open. If the palette is closed it will highlight the palette. Is it possible to open closed palettes when the shape is found?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
You can open it programmatically when the `filter` event is fired.
```ts
stencil.on('filter', (graph, group, keyword) => {    
   if (graph.getCells().length === 0 || !keyword) return;
   stencil.openGroup(group);     
});
```
